packages/sitecore-jss-cli/src/scripts/extract-components.ts

```diff
@@ -0,0 +1,150 @@
+
+/**
+ * @param {Argv} yargs
+ */
+export function args(yargs: Argv) {
+  return yargs.option('appFolder', {
+    requiresArg: false,
+    type: 'string',
+    describe: 'Path to app folder to get components from',
+  });
+}
+
+/**
+ * @param {Argv} yargs
+ */
+export default function builder(yargs: Argv) {
+  return yargs.command(
+    'extract-components',
+    'Reads component files and posts their code for AI learning',
+    args,
+    handler
+  );
+}
+
+/**
+ * Handler fo the extract-component API command
+ * Reads imports from the componentBuilder.ts file and posts the code to the mesh endpoint
+ * @param {object} args - The arguments passed to the command, with optional appName string
+ * @returns {Promise<void>} void
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function handler(args: any) {
+  // TODO: add alternative consent procedure when refining later
+  if (!process.env.EXTRACT_CONSENT) {
+    console.log(chalk.yellow('Skipping code extraction, EXTRACT_CONSENT is not set'));
+    return;
+  }
+  const basePath = args.appFolder ? resolveAppPath(args.appFolder) : process.cwd();
+  try {
+    const componentPaths = resolveImportFiles(basePath);
+    const bearer = await fetchBearerToken();
+    if (!bearer) {
+      console.error(chalk.red('Failed to get bearer token, aborting code extraction'));
+      return;
+    }
+
+    for (const componentPath of componentPaths!) {
+      await sendCode(componentPath, bearer);
+    }
+  } catch (error) {
+    console.error(chalk.red('Error during component extraction:', error));
+  }
+}
+
+const resolveAppPath = (appFolder: string) => {
+  if (path.isAbsolute(appFolder)) return appFolder;
+  return path.resolve(process.cwd(), appFolder);
+};
+
+export const fetchBearerToken = async () => {
+  try {
+    // TODO:adjust when M2M endpoint is live
+    const authenticateResponse = await fetch(M2M_ENDPOINT, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      body: JSON.stringify({
+        client_id: process.env.M2M_CLIENT_ID,
+        client_secret: process.env.M2M_CLIENT_SECRET,
+        audience: process.env.M2M_AUDIENCE,
+        grant_type: 'client_credentials',
+      }),
+    });
+    const jsonResponse = await authenticateResponse.json();
+    return jsonResponse.access_token;
+  } catch (error) {
+    console.error(chalk.red('Error authenticating with M2M token endpoint:', error));
+    return null;
+  }
+};
+
+export const resolveImportFiles = (appPath: string) => {
+  const tsConfig = readConfigFile(path.resolve(appPath, 'tsconfig.json'), ts.sys.readFile);
+
+  if (tsConfig.error) {
+    throw new Error(`Error reading tsconfig.json from JSS app root: ${tsConfig.error.messageText}`);
+  }
+
+  const tsOptions = tsConfig.config.compilerOptions;
+
+  const componentBuilderPath = path.resolve(appPath, 'src', 'temp', 'componentBuilder.ts');
```

Comment:
I think we should have a way to override a default location of component builder file

Reply:
for v2 we would need that, true. For v1 it will always be in src/temp. With that in mind, do we still need it?

Reply:
can users control the name of component factory by changing scripts? if not - we don't need this parameter